pkg/metrics/wal/wal.go

```diff
@@ -122,30 +151,30 @@ type Storage struct {
 	appenderPool sync.Pool
 	bufPool      sync.Pool
 
-	ref    *atomic.Uint64
 	series *stripeSeries
 
 	deletedMtx sync.Mutex
 	deleted    map[chunks.HeadSeriesRef]int // Deleted series, and what WAL segment they must be kept until.
 
 	metrics *storageMetrics
+
+	refIdSource *RefIdSource
 }
 
-// NewStorage makes a new Storage.
-func NewStorage(logger log.Logger, registerer prometheus.Registerer, path string) (*Storage, error) {
+func NewStorageWithRefidSource(logger log.Logger, registerer prometheus.Registerer, path string, refid *RefIdSource) (*Storage, error) {
```

Comment:
Do we want to add this new constructor API?  It seems like it gives consumers a way to re-create the initial problem by creating storages with different `RefIdSources`.  Couldn't we use the `GlobalRefId` in `NewStorage`?

Reply:
At some point, once we are more comfortable with the changes we could switch to using the GlobalRefID.

Reply:
that makes sense -- in that case instead of passing in a `RefIdSource` parameter maybe just have this function use the `GlobalRefID`?

Reply:
Either way here, we're changing the code path for the existing Agent, I'm just trying to avoid adding an API that makes it easy to shoot ourselves in the foot later while being hard to debug.

Reply:
Ended up simplifying it, for this initial pass we dont need the wrapper around atomic, so now just passing around atomics for even less change. Still have the two constructors but hopefully can eventually narrow those down.